test/tls13encryptiontest.c

```diff
@@ -0,0 +1,304 @@
+        "467d99a807dbf778e6ffd8be52456c70665f890811ef2f3c495d5bbe983feedab0c25"
+        "1dde596bc7e2b135909ec9f9166fb0152e8c16a84e4b1039256467f9538be4463",
+        "3381f6b3f94500f16226de440193e858",
+        "4f1d73cc1d465eb30021c41f",
+        "0000000000000001"
+    },
+    {
+        "010015",
+        "6bdf60847ba6fb650da36e872adc684a4af2e8",
+        "eb23a804904b80ba4fe8399e09b1ce42",
+        "efa8c50c06b9c9b8c483e174",
+        "0000000000000001"
+    },
+    {
+        "010015",
+        "621b7cc1962cd8a70109fee68a52efedf87d2e",
+        "3381f6b3f94500f16226de440193e858",
+        "4f1d73cc1d465eb30021c41f",
+        "0000000000000002"
+    }
+};
+
+static int load_record(SSL3_RECORD *rec, size_t recnum, unsigned char **key,
+                       unsigned char *iv, size_t ivlen, unsigned char *seq)
+{
+    unsigned char *pt = NULL, *sq = NULL, *ivtmp = NULL;;
+    long ptlen;
+
+    *key = OPENSSL_hexstr2buf(refdata[recnum].key, NULL);
+    ivtmp = OPENSSL_hexstr2buf(refdata[recnum].iv, NULL);
+    sq = OPENSSL_hexstr2buf(refdata[recnum].seq, NULL);
+    pt = OPENSSL_hexstr2buf(refdata[recnum].plaintext, &ptlen);
+
+    if (*key == NULL || ivtmp == NULL || sq == NULL || pt == NULL)
+        goto err;
+
+    rec->data = rec->input = OPENSSL_malloc(ptlen + EVP_GCM_TLS_TAG_LEN);
+
+    if (rec->data == NULL)
+        goto err;
+
+    rec->length = ptlen;
+    memcpy(rec->data, pt, ptlen);
+    OPENSSL_free(pt);
+    memcpy(seq, sq, SEQ_NUM_SIZE);
+    OPENSSL_free(sq);
+    memcpy(iv, ivtmp, ivlen);
+    OPENSSL_free(ivtmp);
+
+    return 1;
+ err:
+    OPENSSL_free(*key);
+    *key = NULL;
+    OPENSSL_free(ivtmp);
+    OPENSSL_free(sq);
+    OPENSSL_free(pt);
+    return 0;
+}
+
+static int test_record(SSL3_RECORD *rec, size_t recnum, int enc)
+{
+    int ret = 0;
+    unsigned char *refd;
+    long refdatalenl;
+    size_t refdatalen;
+
+    if (enc)
+        refd = OPENSSL_hexstr2buf(refdata[recnum].ciphertext, &refdatalenl);
+    else
+        refd = OPENSSL_hexstr2buf(refdata[recnum].plaintext, &refdatalenl);
+
+    if (refd == NULL) {
+        fprintf(stderr, "Failed to get reference data for record %"OSSLzu
+                         "\n", recnum);
+        goto err;
+    }
+    refdatalen = (size_t)refdatalenl;
+
+    if (rec->length != refdatalen) {
+        fprintf(stderr, "Unexpected length for record %"OSSLzu"\n", recnum);
+        goto err;
+    }
+
+    if (memcmp(rec->data, refd, refdatalen) != 0) {
+        fprintf(stderr, "Data does not match for record %"OSSLzu"\n", recnum);
+        goto err;
+    }
+
+    ret = 1;
+
+ err:
+    OPENSSL_free(refd);
+    return ret;
+}
+static int test_tls13_encryption(void)
+{
+    SSL_CTX *ctx = NULL;
+    SSL *s = NULL;
+    SSL3_RECORD rec;
+    unsigned char *key = NULL, *iv = NULL, *seq = NULL;
+    const EVP_CIPHER *ciph = EVP_aes_128_gcm();
+    int ret = 0;
+    size_t ivlen, ctr;
+
+    rec.data = NULL;
+
+    ctx = SSL_CTX_new(TLS_method());
```

Comment:
No we don't!!!! There are examples of that, sure. But there are plenty of examples where we don't. It is definitely not our standard, and it isn't in our style guide. There are occasions where it makes sense, but those are rare and this isn't one of them.